### PR TITLE
Material: Make .vertexColors a boolean.

### PR DESCRIPTION
**Pull request by Mugen87**
This PR makes `Material.vertexColors` to a boolean. However, it's still possible for backwards compatibility reasons to use the old style by assigning one of the constants `THREE.NoColors`, `THREE.VertexColors` and `THREE.FaceColors`.

More information: https://github.com/mrdoob/three.js/pull/14788#issuecomment-416054514

### CODE REVIEW

**Comment by donmccurdy** [comment]
Agreed on this change!

Do you want the `strict === true` checks in this PR, though? Just looking for truthiness would be more backward-compatible.

**Comment by Mugen87** [comment]
> Just looking for truthiness would be more backward-compatible.

Makes totally sense! I've changed `Projector` and `SVGRenderer` accordingly.

**Review by donmccurdy — APPROVED** [review]

**Comment by donmccurdy** [comment]
Does `Geometry` + `FaceColors` do anything a this point? With the former deprecated I don't think we want to keep `FaceColors` in the `Material` API, but just want to know what to expect for the next release.

**Comment by Mugen87** [comment]
> Does Geometry + FaceColors do anything a this point?

At least not in context of `WebGLRenderer`. It's just important where you define color values in `Face3`.

https://github.com/mrdoob/three.js/blob/76d16bd828c8d3e1870eac45aa466c20313cf944/src/core/DirectGeometry.js#L175-L187

`SVGRenderer` and `Projector` were the last classes performing checks on `THREE.FaceColors`. However, both threat `THREE.FaceColors` and `THREE.VertexColors` equally since a while. They do not really support vertex colors because of missing color interpolation. They expect vertex color values to be equal for a single triangle.

**Comment by Mugen87** [comment]
TBH, I would remove `Projector` and `SVGRenderer` at some point. Especially since `SVGLoader` can be used now to render SVG files with WebGL.

**Comment by donmccurdy** [comment]
> At least not in context of WebGLRenderer. It's just important where you define color values in Face3.

Ok, thanks!

File: src/materials/Material.js [hunk 1]
```diff
@@ -249,7 +249,7 @@ Material.prototype = Object.assign( Object.create( EventDispatcher.prototype ),
 		if ( this.blending !== NormalBlending ) data.blending = this.blending;
 		if ( this.flatShading === true ) data.flatShading = this.flatShading;
 		if ( this.side !== FrontSide ) data.side = this.side;
-		if ( this.vertexColors !== NoColors ) data.vertexColors = this.vertexColors;
+		if ( this.vertexColors === true || this.vertexColors > 0 ) data.vertexColors = true;
```
Comment (mrdoob) [inline src/materials/Material.js]: Do we need ` || this.vertexColors > 0` here?

**Comment by mrdoob** [comment]
Nice clean up!

**Comment by mrdoob** [comment]
> TBH, I would remove `Projector` and `SVGRenderer` at some point. Especially since `SVGLoader` can be used now to render SVG files with WebGL.

Well, the idea of `SVGRenderer` is to produce SVGs, which I've seem people using for creating vector illustrations for books.

File: src/materials/Material.js (hunk 1 above)
Comment (mrdoob) [inline src/materials/Material.js]: Do we need ` || this.vertexColors > 0` here?

Comment (Mugen87) [reply src/materials/Material.js]: I think if we change the check to just `vertexColors`, it should be also work. Let me make this change with a new commit 👍

**Comment by Mugen87** [comment]
> Well, the idea of SVGRenderer is to produce SVGs, which I've seem people using for creating vector illustrations for books.

Okay, that's of course a valid use case for the renderer.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**